pkgs/development/python-modules/funcparserlib/default.nix

```diff
@@ -1,31 +1,52 @@
 { lib
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
+, fetchpatch
+, poetry-core
 , python
-, isPy3k
+, pytestCheckHook
+, pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "funcparserlib";
-  version = "0.3.6";
+  version = "1.0.0a0";
+  format = "pyproject";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "b7992eac1a3eb97b3d91faa342bfda0729e990bd8a43774c1592c091e563c91d";
+  disabled = pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "vlasovskikh";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-YfcboKjyc5ASzrp0duu2R6psf51MGZIeZ0owo5QNSnU=";
   };
 
-  checkPhase = ''
-    ${python.interpreter} -m unittest discover
-  '';
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
 
-  # Tests are Python 2.x only judging from SyntaxError
-  doCheck = !(isPy3k);
+  patches = [
+    # Support for poetry-core, https://github.com/vlasovskikh/funcparserlib/pull/73
+    (fetchpatch {
+      name = "support-poetry-core.patch";
+      url = "https://github.com/vlasovskikh/funcparserlib/commit/61ed558fc146b7a30879919325dfa8aae77be556.patch";
```

Comment:
IMHO, given that the PR hasn't been merged yet, I don't think it's a good idea to reference it in nixpkgs. What if the PR gets rebased (or closed)?

Reply:
> What if the PR gets rebased (or closed)?

Then we need to adjust it. We want that packages depend on `poetry-core` instead of `poetry`. In most of the cases `poetry-core` is a drop-in replacement. Thus, we could just patch it in nixpkgs but this way upstream wouldn't profit it.

Reply:
> > What if the PR gets rebased (or closed)?
> 
> Then we need to adjust it.

But it breaks building old versions. IMHO, we should strive for using stable URLs. Once upstream merges the pr, you can refer to the (merged) upstream commit.

Reply:
> But it breaks building old versions.

The previous release (0.3.6) was made over 8 years ago and 1.0.0 will freeze that [API](https://github.com/vlasovskikh/funcparserlib/blob/master/docs/changes.md). This means that 0.3.6 doesn't have to be build. 

> Once upstream merges the pr, you can refer to the (merged) upstream commit.

It's common practice to point to commits and not to the commits which are part of the PR.

Reply:
I'm talking about "older" nixpkgs versions. I think of PR URLs like non-permalinks, that can break at any time. Anyway, I'm not blocking this but personally not approving either.

Reply:
I think nixpkgs is a software archive, and I rely on this to do git bisect. Non-permalink URLs effectively break git bisect.

Reply:
referencing a commit directly is stable (not sure if this was changed by the time I took a look)

Reply:
That URL is in practice permanent. You can always get the file from the nixos cache when it once hit nixos-unstable.